### ISSUE TITLE
AssertionError assert(self._name is not None)

### ISSUE THREAD

**Issue by chris48s** [jstrieb/github-stats#5]
Hello. I tried forking this and I'm getting builds failing with

```
Python 3.8.6
Traceback (most recent call last):
  File "generate_images.py", line 107, in <module>
    asyncio.run(main())
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "generate_images.py", line 103, in main
    await asyncio.gather(generate_languages(s), generate_overview(s))
  File "generate_images.py", line 36, in generate_overview
    output = re.sub("{{ name }}", await s.name, output)
  File "/home/runner/work/github-stats/github-stats/github_stats.py", line 347, in name
    assert(self._name is not None)
AssertionError

Error: Process completed with exit code 1.
```

example https://github.com/chris48s/github-stats/runs/1250004507

Any thoughts on what may be going wrong? I had a look through a few other forks and I haven't spotted anyone else with the same issue.

**Comment by jstrieb** [comment]
Hi, thanks for taking a look at the project! I've been pretty busy (hence the delay in getting back about this issue), but just had a chance to look at this.

The issue you point out is not something I've seen before, and after staring at the code for several minutes, I'm having trouble seeing how that could even happen. I've included the relevant snippet of code below, and the only way I could see this happening is if the GitHub API query results contain a `None` value for the `name` key.

https://github.com/jstrieb/github-stats/blob/fc2a7e42c313ca34d60dbcda3cb354ad04b44de2/github_stats.py#L268-L289

Note that unless the GitHub GraphQL API query response is malformed, there should be a non-null location at that value. Either the API is misbehaving, or I misunderstand their GraphQL API and its failure modes. See below for the relevant part of the actual query that is being made, which explicitly includes the required value.

https://github.com/jstrieb/github-stats/blob/fc2a7e42c313ca34d60dbcda3cb354ad04b44de2/github_stats.py#L101-L110

I will continue to investigate, but at the moment, I am kind of at a loss regarding what could be going wrong. I will reply to this issue if I discover anything new or issue a patch.

**Comment by chris48s** [comment]
I see. If I run

```gql
query { 
  viewer {
    name
  }
}
```

in the GraphQL explorer,  it returns

```json
{
  "data": {
    "viewer": {
      "name": null
    }
  }
}
```

so.. that explains that. I guess that's because on my account I don't actually have a "real" name set - just my username.

If I call

```gql
query { 
  viewer {
    login,
    name
  }
}
```

I get back

```json
{
  "data": {
    "viewer": {
      "login": "chris48s",
      "name": null
    }
  }
}
```

..so you could query that and fall back to `login` if `name` is null

**Comment by jstrieb** [comment]
Attempted to add that functionality! Never occurred to me that the name could be null, though I may have not read the API documentation carefully enough. 

Thanks for looking into this and presenting the solution in a way that was easy for me to apply a fix. I've made a PR on your fork of the repo with the proposed fix. I tested locally and it worked for me, but I could not replicate a null value for that key in the response, so I'm not 100% sure it will work.
